tools/shell-commands/src/main/java/org/apache/unomi/shell/migration/impl/MigrationTo200.java

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.unomi.shell.migration.impl;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.unomi.shell.migration.Migration;
+import org.json.JSONObject;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Version;
+import org.osgi.service.component.annotations.Component;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class MigrationTo200 implements Migration {
+
+    private CloseableHttpClient httpClient;
+    private Session session;
+    private String esAddress;
+
+    @Override
+    public Version getFromVersion() {
+        return new Version("1.5.0");
+    }
+
+    @Override
+    public Version getToVersion() {
+        return new Version("2.0.0");
+    }
+
+    @Override
+    public String getDescription() {
+        return "Updates mapping for an index with prefix \"context-event\". Adds the \"sourceId\" field and copies value from the \"scope\" field to it.";
+    }
+
+    @Override
+    public void execute(Session session, CloseableHttpClient httpClient, String esAddress, BundleContext bundleContext) throws IOException {
+        this.httpClient = httpClient;
+        this.session = session;
+        this.esAddress = esAddress;
+
+        doExecute();
+    }
+
+    private void doExecute() throws IOException {
+        try (CloseableHttpResponse response = httpClient.execute(new HttpGet(esAddress + "/_aliases"))) {
+
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                JSONObject indicesAsJson = new JSONObject(EntityUtils.toString(response.getEntity()));
+
+                final Set<String> indices = indicesAsJson.keySet().stream().
+                        filter(alias -> alias.startsWith("context-event")).
+                        collect(Collectors.toSet());
+
+                for (String indexName : indices) {
+                    updateMapping(indexName, httpClient);
+                }
+            }
+        }
+    }
+
+    private void updateMapping(final String indexName, final CloseableHttpClient httpClient) throws IOException {
+        HttpPut httpPut = new HttpPut(esAddress + "/" + indexName + "/_mapping");
+
+        httpPut.addHeader("Accept", "application/json");
+        httpPut.addHeader("Content-Type", "application/json");
+
+        String request = "{\n" +
+                "\"properties\": {\n" +
+                " \"sourceId\": {\n" +
```

Comment:
Seems like the analyzer is missing here.

Reply:
fixed